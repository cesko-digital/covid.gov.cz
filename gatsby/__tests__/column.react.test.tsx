import React from 'react';
import renderer from 'react-test-renderer';

import Col from '@/components/col';

test('Column renders correctly', () => {
  const component = renderer.create(<Col />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Column with content renders correctly', () => {
  const component = renderer.create(<Col>content</Col>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Column with responsive column classes renders correctly', () => {
  const component = renderer.create(
    <Col col={11} colSm={12} colMd={9} colLg={6} colXl={6} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Column with additional class names renders correctly', () => {
  const component = renderer.create(<Col className="class" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Column with available align-self properties renders correctly', () => {
  const component = renderer.create(
    <>
      <Col alignSelf="start" />
      <Col alignSelf="center" />
      <Col alignSelf="end" />
    </>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
