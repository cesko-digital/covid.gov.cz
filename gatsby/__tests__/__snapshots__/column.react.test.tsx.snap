// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders correctly 1`] = `
<div
  className=""
/>
`;

exports[`Column with additional class names renders correctly 1`] = `
<div
  className="class"
/>
`;

exports[`Column with available align-self properties renders correctly 1`] = `
Array [
  <div
    className="align-self-start"
  />,
  <div
    className="align-self-center"
  />,
  <div
    className="align-self-end"
  />,
]
`;

exports[`Column with content renders correctly 1`] = `
<div
  className=""
>
  content
</div>
`;

exports[`Column with responsive column classes renders correctly 1`] = `
<div
  className="col-11 col-sm-12 col-md-9 col-lg-6 col-xl-6"
/>
`;
